Source/Core/DolphinQt/Config/GameConfigWidget.cpp

```diff
@@ -196,207 +192,171 @@ void GameConfigWidget::CreateWidgets()
+      // ini being edited.
+      for (int i = 0; i < m_local_tab->count(); i++)
+      {
+        if (m_local_tab->tabText(i).toStdString() == m_game_id + ".ini")
+        {
+          m_local_tab->removeTab(i);
+
+          auto* edit = new GameConfigEdit(
+              nullptr,
+              QString::fromStdString(File::GetUserPath(D_GAMESETTINGS_IDX) + m_game_id + ".ini"),
+              false);
 
-void GameConfigWidget::ConnectWidgets()
-{
-  // Buttons
-  connect(m_refresh_config, &QPushButton::clicked, this, &GameConfigWidget::LoadSettings);
-
-  for (QCheckBox* box :
-       {m_enable_dual_core, m_enable_mmu, m_enable_fprf, m_sync_gpu, m_emulate_disc_speed,
-        m_use_dsp_hle, m_manual_texture_sampling, m_use_monoscopic_shadows})
-    connect(box, &QCheckBox::stateChanged, this, &GameConfigWidget::SaveSettings);
-
-  connect(m_deterministic_dual_core, &QComboBox::currentIndexChanged, this,
-          &GameConfigWidget::SaveSettings);
-  connect(m_depth_slider, &QSlider::valueChanged, this, &GameConfigWidget::SaveSettings);
-  connect(m_convergence_spin, &QSpinBox::valueChanged, this, &GameConfigWidget::SaveSettings);
-}
+          m_local_tab->insertTab(i, edit, QString::fromStdString(m_game_id + ".ini"));
+          break;
+        }
+      }
+    }
 
-void GameConfigWidget::LoadCheckBox(QCheckBox* checkbox, const std::string& section,
-                                    const std::string& key, bool reverse)
-{
-  bool checked;
-  if (m_gameini_local.GetOrCreateSection(section)->Get(key, &checked))
-    return checkbox->setCheckState(checked ^ reverse ? Qt::Checked : Qt::Unchecked);
+    // Update other tabs after using ini editor.
+    if (m_prev_tab_index == editor_index)
+    {
+      // Load won't clear deleted keys, so everything is wiped before loading.
+      m_layer->DeleteAllKeys();
+      m_layer->Load();
+      Config::OnConfigChanged();
+    }
 
-  if (m_gameini_default.GetOrCreateSection(section)->Get(key, &checked))
-    return checkbox->setCheckState(checked ^ reverse ? Qt::Checked : Qt::Unchecked);
-  checkbox->setCheckState(Qt::PartiallyChecked);
+    m_prev_tab_index = index;
+  });
+
+  auto help_icon = style()->standardIcon(QStyle::SP_MessageBoxQuestion);
+  auto* help_label = new QLabel(tr("These settings override core Dolphin settings."));
+  help_label->setToolTip(
+      tr("Italics mark a default game setting, bold marks user "
+         "settings.\nRight-click to remove user settings.\n AA settings are disabled when the "
+         "global backend doesn't match the game setting."));
+  auto help_label_icon = new QLabel();
+  help_label_icon->setPixmap(help_icon.pixmap(12, 12));
+  help_label_icon->setToolTip(
+      tr("Italics mark a default game setting, bold marks user "
+         "settings.\nRight-click to remove user settings.\n AA settings are disabled when the "
+         "global backend doesn't match the game setting."));
```

Comment:
Please call `tr` just once and then reuse the result. Having two copies of the same string causes a risk that someone will edit one but forget to edit the other.

Reply:
Does my method look right? Don't want to mess up the TRs.

Reply:
Yes, it looks good.